### PR TITLE
Added solution for Project Euler problem 234

### PR DESCRIPTION
**Pull request by fpringle**
### Name: Semidivisible numbers

For an integer n ≥ 4, we define the lower prime square root of n, denoted by lps(n), as the largest prime ≤ √n and the upper prime square root of n, ups(n), as the smallest prime ≥ √n.

So, for example, lps(4) = 2 = ups(4), lps(1000) = 31, ups(1000) = 37.
Let us call an integer n ≥ 4 semidivisible, if one of lps(n) and ups(n) divides n, but not both.

The sum of the semidivisible numbers not exceeding 15 is 30, the numbers are 8, 10 and 12.
15 is not semidivisible because it is a multiple of both lps(15) = 3 and ups(15) = 5.
As a further example, the sum of the 92 semidivisible numbers up to 1000 is 34825.

What is the sum of all semidivisible numbers not exceeding 999966663333 ?

### Note:
My solution is very similar at its core to sol1.py. Mine removes a few loops that slow down the algorithm and adds a nice prime iterator.

Reference: [https://projecteuler.net/problem=234](url)
Reference: #2695

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Review by dhruvmanila — COMMENTED** [review]
### Announcement:

@fpringle

*This message is sent to all pull requests which are **OPEN, NOT MERGED, not containing 'wontfix' label and contains 'Project Euler' or 'Project' or 'Euler' (case doesn't matter) in their title**. If this message doesn't apply to your pull request, please ignore this.*

### Message:

This is to notify all the users submitting a pull request for the Project Euler solution that the [README.md](https://github.com/TheAlgorithms/Python/blob/master/project_euler/README.md) file has been updated with all the style requirements necessary for the directory and solution files. ***Please read through it and make all the necessary changes required to your submission.***

File: project_euler/problem_234/sol2.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    for i in range(3, limit + 1, 2):
+        if not primes[i]:
+            continue
+        for j in range(i * i, limit + 1, i):
+            primes[j] = False
+
+    return itertools.compress(itertools.count(), primes)
+
+
+def pairwise(iterable: Iterator[int]) -> Iterator[tuple]:
```
Comment (dhruvmanila) [inline project_euler/problem_234/sol2.py:50]:
```suggestion
def pairwise(iterable: Iterator[int]) -> Iterator[tuple[int]]:
```
Is that correct?
Also, do you want an `Iterator` or an `Iterable` as your input?

**Comment by dhruvmanila** [comment]
Please leave a newline between the docstring and `doctest`

File: project_euler/problem_234/sol2.py (hunk 1 above)
Comment (fpringle) [reply project_euler/problem_234/sol2.py:50]: Absolutey right, thanks!

**Pull request closed without merging**